### PR TITLE
jsoncpp 1.9.3

### PR DESCRIPTION
**Pull request by igas**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by igas** [comment]
Only redex fails on highsierra. jsoncpp is hardcoded here: https://github.com/facebook/redex/blob/7c64a5d0e333f92b0f2a1419ff18b45451a7a579/Makefile.am#L111

If someone knows if we can overwrite it through flags let me know. As I'm not that familiar with these building systems.

**Comment by SMillerDev** [comment]
Just inreplace and make an upstream issue?

File: Formula/lgogdownloader.rb [hunk 1]
```diff
@@ -23,7 +23,8 @@ class Lgogdownloader < Formula
   depends_on "tinyxml2"
 
   def install
-    system "cmake", ".", *std_cmake_args
+    system "cmake", ".", "-DJSONCPP_INCLUDE_DIR=#{Formula["libffi"].include}",
```
Comment (bayandin) [inline Formula/lgogdownloader.rb:26]:
```suggestion
    system "cmake", ".", "-DJSONCPP_INCLUDE_DIR=#{Formula["jsoncpp"].include}",
```

**Comment by bayandin** [comment]
It turns out that `jsoncpp` has only prereleases since [0.10.7 (December 2018)](https://github.com/open-source-parsers/jsoncpp/releases).

And there's a build failure on High Sierra:
```
2020-04-23T22:51:23.0907680Z 	/bin/sh ./libtool  --tag=CXX   --mode=compile clang++ -DPACKAGE_NAME=\"redex\" -DPACKAGE_TARNAME=\"redex\" -DPACKAGE_VERSION=\"1.0\" -DPACKAGE_STRING=\"redex\ 1.0\" -DPACKAGE_BUGREPORT=\"not-valid-yet@fb.com\" -DPACKAGE_URL=\"\" -DPACKAGE=\"redex\" -DVERSION=\"1.0\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DPACKAGE=\"redex\" -DVERSION=\"1.0\" -DHAVE_PTHREAD_PRIO_INHERIT=1 -DHAVE_PTHREAD=1 -DHAVE_BOOST=/\*\*/ -DHAVE_BOOST_FILESYSTEM=/\*\*/ -DHAVE_BOOST_SYSTEM=/\*\*/ -DHAVE_BOOST_REGEX=/\*\*/ -DHAVE_BOOST_PROGRAM_OPTIONS=/\*\*/ -DHAVE_LIBZ=1 -DHAVE_LIBJSONCPP=1 -DHAVE_ARPA_INET_H=1 -DHAVE_FCNTL_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_MEMORY_H=1 -DHAVE_NETINET_IN_H=1 -DHAVE_STDDEF_H=1 -DHAVE_STDINT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_UNISTD_H=1 -DHAVE__BOOL=1 -DHAVE_STDLIB_H=1 -DHAVE_MALLOC=1 -DHAVE_STDLIB_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_PARAM_H=1 -DHAVE_GETPAGESIZE=1 -DHAVE_MMAP=1 -DHAVE_STDLIB_H=1 -DHAVE_REALLOC=1 -DHAVE_CLOCK_GETTIME=1 -DHAVE_GETTIMEOFDAY=1 -DHAVE_MEMMOVE=1 -DHAVE_MEMSET=1 -DHAVE_MUNMAP=1 -DHAVE_REGCOMP=1 -DHAVE_STRCHR=1 -DHAVE_STRDUP=1 -DHAVE_STRERROR=1 -DHAVE_STRRCHR=1 -DHAVE_STRSTR=1 -DHAVE_STRTOL=1 -I.  -I./liblocator -I./libredex -I./libresource -I./libresource/android -I./libresource/androidfw -I./libresource/cutils -I./libresource/system -I./libresource/utils -I./opt -I./opt/access-marking -I./opt/add_redex_txt_to_apk -I./opt/analysis_ref_graph -I./opt/annoclasskill -I./opt/annokill -I./opt/bridge -I./opt/check_breadcrumbs -I./opt/constant_propagation -I./opt/dedup_blocks -I./opt/delinit -I./opt/delsuper -I./opt/final_inline -I./opt/hotness-score -I./opt/inlineinit -I./opt/interdex -I./opt/local-dce -I./opt/obfuscate -I./opt/original_name -I./opt/outliner -I./opt/peephole -I./opt/print-members -I./opt/rebindrefs -I./opt/redundant_move_elimination -I./opt/regalloc -I./opt/remove-builders -I./opt/remove-unreachable -I./opt/remove_empty_classes -I./opt/remove_gotos -I./opt/renameclasses -I./opt/reorder-interfaces -I./opt/shorten-srcstrings -I./opt/simpleinline -I./opt/singleimpl -I./opt/static-sink -I./opt/staticrelo -I./opt/string_simplification -I./opt/strip-debug-info -I./opt/synth -I./opt/test_cfg -I./opt/track_resources -I./opt/unreferenced_interfaces -I./opt/unterface -I./opt/verifier -I./opt/virtual_scope -I./tools/common -I./tools/redexdump -I./util -I/usr/local/opt/jsoncpp/include  --std=gnu++14 -O3 -Wall -g  -MT libresource/String8.lo -MD -MP -MF $depbase.Tpo -c -o libresource/String8.lo libresource/String8.cpp &&\
2020-04-23T22:51:23.0909280Z 	mv -f $depbase.Tpo $depbase.Plo
2020-04-23T22:51:23.0909990Z libresource/RedexResources.cpp:216:8: error: no member named 'stack' in namespace 'std'
2020-04-23T22:51:23.0910180Z   std::stack<android::Res_value> nodes_to_explore;
2020-04-23T22:51:23.0910290Z   ~~~~~^
2020-04-23T22:51:23.0910980Z libresource/RedexResources.cpp:216:32: error: expected '(' for function-style cast or type construction
2020-04-23T22:51:23.0911370Z   std::stack<android::Res_value> nodes_to_explore;
2020-04-23T22:51:23.0911490Z              ~~~~~~~~~~~~~~~~~~^
2020-04-23T22:51:23.0912480Z libresource/RedexResources.cpp:216:34: error: use of undeclared identifier 'nodes_to_explore'
2020-04-23T22:51:23.0912680Z   std::stack<android::Res_value> nodes_to_explore;
2020-04-23T22:51:23.0912770Z                                  ^
2020-04-23T22:51:23.0913500Z libresource/RedexResources.cpp:218:5: error: use of undeclared identifier 'nodes_to_explore'
2020-04-23T22:51:23.0913650Z     nodes_to_explore.push(initial_values[index]);
2020-04-23T22:51:23.0913760Z     ^
2020-04-23T22:51:23.0914420Z libresource/RedexResources.cpp:221:11: error: use of undeclared identifier 'nodes_to_explore'
2020-04-23T22:51:23.0914570Z   while (!nodes_to_explore.empty()) {
2020-04-23T22:51:23.0914670Z           ^
2020-04-23T22:51:23.0915350Z libresource/RedexResources.cpp:222:28: error: use of undeclared identifier 'nodes_to_explore'
2020-04-23T22:51:23.0915500Z     android::Res_value r = nodes_to_explore.top();
2020-04-23T22:51:23.0915610Z                            ^
2020-04-23T22:51:23.0916300Z libresource/RedexResources.cpp:223:5: error: use of undeclared identifier 'nodes_to_explore'
2020-04-23T22:51:23.0916450Z     nodes_to_explore.pop();
2020-04-23T22:51:23.0916530Z     ^
2020-04-23T22:51:23.0917210Z libresource/RedexResources.cpp:243:7: error: use of undeclared identifier 'nodes_to_explore'
2020-04-23T22:51:23.0917360Z       nodes_to_explore.push(inner_values[index]);
2020-04-23T22:51:23.0917450Z       ^
2020-04-23T22:51:23.0921620Z libtool: compile:  clang++ -DPACKAGE_NAME=\"redex\" -DPACKAGE_TARNAME=\"redex\" -DPACKAGE_VERSION=\"1.0\" "-DPACKAGE_STRING=\"redex 1.0\"" -DPACKAGE_BUGREPORT=\"not-valid-yet@fb.com\" -DPACKAGE_URL=\"\" -DPACKAGE=\"redex\" -DVERSION=\"1.0\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DPACKAGE=\"redex\" -DVERSION=\"1.0\" -DHAVE_PTHREAD_PRIO_INHERIT=1 -DHAVE_PTHREAD=1 "-DHAVE_BOOST=/**/" "-DHAVE_BOOST_FILESYSTEM=/**/" "-DHAVE_BOOST_SYSTEM=/**/" "-DHAVE_BOOST_REGEX=/**/" "-DHAVE_BOOST_PROGRAM_OPTIONS=/**/" -DHAVE_LIBZ=1 -DHAVE_LIBJSONCPP=1 -DHAVE_ARPA_INET_H=1 -DHAVE_FCNTL_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_MEMORY_H=1 -DHAVE_NETINET_IN_H=1 -DHAVE_STDDEF_H=1 -DHAVE_STDINT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_UNISTD_H=1 -DHAVE__BOOL=1 -DHAVE_STDLIB_H=1 -DHAVE_MALLOC=1 -DHAVE_STDLIB_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_PARAM_H=1 -DHAVE_GETPAGESIZE=1 -DHAVE_MMAP=1 -DHAVE_STDLIB_H=1 -DHAVE_REALLOC=1 -DHAVE_CLOCK_GETTIME=1 -DHAVE_GETTIMEOFDAY=1 -DHAVE_MEMMOVE=1 -DHAVE_MEMSET=1 -DHAVE_MUNMAP=1 -DHAVE_REGCOMP=1 -DHAVE_STRCHR=1 -DHAVE_STRDUP=1 -DHAVE_STRERROR=1 -DHAVE_STRRCHR=1 -DHAVE_STRSTR=1 -DHAVE_STRTOL=1 -I. -I./liblocator -I./libredex -I./libresource -I./libresource/android -I./libresource/androidfw -I./libresource/cutils -I./libresource/system -I./libresource/utils -I./opt -I./opt/access-marking -I./opt/add_redex_txt_to_apk -I./opt/analysis_ref_graph -I./opt/annoclasskill -I./opt/annokill -I./opt/bridge -I./opt/check_breadcrumbs -I./opt/constant_propagation -I./opt/dedup_blocks -I./opt/delinit -I./opt/delsuper -I./opt/final_inline -I./opt/hotness-score -I./opt/inlineinit -I./opt/interdex -I./opt/local-dce -I./opt/obfuscate -I./opt/original_name -I./opt/outliner -I./opt/peephole -I./opt/print-members -I./opt/rebindrefs -I./opt/redundant_move_elimination -I./opt/regalloc -I./opt/remove-builders -I./opt/remove-unreachable -I./opt/remove_empty_classes -I./opt/remove_gotos -I./opt/renameclasses -I./opt/reorder-interfaces -I./opt/shorten-srcstrings -I./opt/simpleinline -I./opt/singleimpl -I./opt/static-sink -I./opt/staticrelo -I./opt/string_simplification -I./opt/strip-debug-info -I./opt/synth -I./opt/test_cfg -I./opt/track_resources -I./opt/unreferenced_interfaces -I./opt/unterface -I./opt/verifier -I./opt/virtual_scope -I./tools/common -I./tools/redexdump -I./util -I/usr/local/opt/jsoncpp/include --std=gnu++14 -O3 -Wall -g -MT libresource/ResourceTypes.lo -MD -MP -MF libresource/.deps/ResourceTypes.Tpo -c libresource/ResourceTypes.cpp -o libresource/ResourceTypes.o >/dev/null 2>&1
2020-04-23T22:51:23.0922400Z 8 errors generated.
```

**Comment by Bo98** [comment]
> It turns out that jsoncpp has only prereleases since 0.10.7 (December 2018).

1.8.4 is also not a pre-release. It was just realeased before 0.10.7. 1.9.0 and 1.9.1 probably shouldn't have been included in Homebrew.

High Sierra error looks like a missing `#include <stack>`. Not sure if jsoncpp had anything to do with that.

**Comment by issyl0** [comment]
Is this still in progress? It's got a few conflicts. There seems to be a newer version of `jsoncpp` since - it's non-prerelease, but 00.x - whatever that means!

**Comment by bayandin** [comment]
Regarding versioning (a quote from `jsoncpp` readme):
> 1.y.z is built with C++11.
0.y.z can be used with older compilers.
00.11.z can be used both in old and new compilers.

So I guess we should stick to 1.y.z releases.
And I also guess, that `1.9.2` shouldn't be merged since it's a prerelease version (unless it fixes something important)

(and the current version (1.9.1) should be whitelisted in prereleases audit)

**Comment by samford** [comment]
Just a heads up, [version 1.9.3 is out](https://github.com/open-source-parsers/jsoncpp/releases/tag/v1.9.3) and it's marked as "latest release" (not pre-release).

**Comment by bayandin** [comment]
> Just a heads up, [version 1.9.3 is out](https://github.com/open-source-parsers/jsoncpp/releases/tag/v1.9.3) and it's marked as "latest release" (not pre-release).

Let's have a look at it. Thanks!

**Comment by bayandin** [comment]
The only fail is:
```
==> brew audit --skip-style sysdig --online
==> FAILED
Error: 1 problem in 1 formula detected
sysdig:
  * The URL https://sysdig.com/ is not reachable
```
And https://sysdig.com/ opens perfectly fine for me.

So I believe we're ready to go 🎉

**Review by bayandin — APPROVED** [review]

**Pull request closed without merging**